Mathlib/RingTheory/LinearDisjoint.lean

```diff
@@ -412,6 +443,86 @@ variable {A B}
 
 variable (H : A.LinearDisjoint B)
 
+include H in
+/-- If `A` and `B` are subalgebras in a domain `S` over `R`, and if they are
+linearly disjoint, then `A ⊗[R] B` is also a domain. -/
+theorem isDomain [IsDomain S] : IsDomain (A ⊗[R] B) :=
+  have := Subtype.val_injective.isDomain (algebraMap ↥(A ⊔ B) S)
+  H.mulMap.toMulEquiv.isDomain
+
+/-- If `A` and `B` are `R`-algebras, such that there exists a domain `S` such that
+`A` and `B` inject into it and their images are linearly disjoint,
+then `A ⊗[R] B` is also a domain. -/
+theorem isDomain_of_injective [IsDomain S] {A B : Type*} [CommRing A] [CommRing B]
```

Comment:
```suggestion
theorem isDomain_of_injective [IsDomain S] {A B : Type*} [Semiring A] [Semiring B]
```
also R and S can be CommSemirings in this one. Whenever flatness is mentioned, though, we must assume Ring. I'm generalizing flatness in #19115, but the current definition is apparently not strong enough for semirings, so I'll need to change the definition ...

Reply:
But `Function.Injective.isDomain` requires `Ring`...

Reply:
OK generalized to `Semiring` in PR #19774. Comments welcome.